docs/content/formatter.md

```diff
@@ -22,242 +22,430 @@ The formatter follows these key principles:
+for (item in collection) {
+  process(item)
 }
-while (TRUE) {
-  foo()
+for (i in 1:length(data)) {
+  calculate(data[i])
+}
+```
+
+**While loops** follow similar block enforcement rules:
+
+```r
+# Before formatting
+while(condition) action()
+
+# After formatting
+while (condition) {
+  action()
 }
+```
+
+**Repeat loops** also enforce braced blocks:
+
+```r
+# Before formatting
+repeat action()
+
+# After formatting
 repeat {
-  foo()
+  action()
 }
 ```
 
+**Loop condition formatting**: Complex conditions that span multiple lines receive proper indentation within the parentheses.
+
 ### Parenthesized Expressions
 
-Parenthesized expressions maintain their layout, with consistent formatting for multi-line expressions:
+Parenthesized expressions maintain their layout with smart formatting for readability:
 
 ```r
 # Before formatting
-(a+b)
-(a +
- b)
+(x+y*z)
+(very_long_expression+
+ another_part)
 
 # After formatting
-(a + b)
+(x + y * z)
 (
-  a + b
+  very_long_expression +
+  another_part
 )
 ```
 
-### Strings
+**Parenthesis hugging**: If the content fits on one line, parentheses hug the content. For multiline content, proper indentation is applied.
+
+### String Literals
+
+String literals receive intelligent quote normalization:
+
+```r
+# Before formatting
+message <- "Hello world"
+quoted_content <- 'Say "hello"'
+
+# After formatting
+message <- "Hello world"
+quoted_content <- 'Say "hello"'
+```
+
+**Quote selection**: The formatter prefers single quotes unless the string content contains unescaped double quotes. This helps avoid unnecessary escaping while maintaining readability.
+
+### Subsetting and Member Access
+
+**Bracket subsetting** follows the same formatting rules as function calls:
+
+```r
+# Before formatting
+data[row_index,column_index]
+matrix[i=1,j=2,drop=FALSE]
+
+# After formatting
+data[row_index, column_index]
+matrix[i = 1, j = 2, drop = FALSE]
+```
+
+**Double bracket subsetting** for list/environment access:
+
+```r
+# Before formatting
+environment[["variable_name"]]
+nested_list[[key1]][[key2]]
+
+# After formatting
+environment[["variable_name"]]
+nested_list[[key1]][[key2]]
+```
+
+**Member access operators** (`$` and `@`):
 
-String literals are consistently formatted, preserving escape sequences and multi-line content:
+```r
+# Before formatting
+object$member_variable
+s4_object@slot_name
+
+# After formatting
+object$member_variable
+s4_object@slot_name
+```
+
+**Namespace operators** (`::` and `:::`):
+
+```r
+# Before formatting
+package::public_function
+package:::private_function
+
+# After formatting
+package::public_function
+package:::private_function
+```
+
+### Unary Operators
+
+Unary operators receive appropriate spacing based on their type and context:
 
 ```r
 # Before formatting
-x <- "hello"
-y <- 'world'
+result=!condition
+number=-42
+formula=~response+predictor
 
 # After formatting
-x <- "hello"
-y <- "world"
+result = !condition
+number = -42
+formula = ~ response + predictor
```

Comment:
unecessary to add no spaces in before formatting. not relevant for that case. add a space before - and ~ in before formatting